src/mudlet-lua/tests/Other_spec.lua

```diff
@@ -1,4 +1,187 @@
+      _G.tempTimer = function(time, code)
+        if type(code) == "string" then
+          loadstring(code)()
+        elseif type(code) == "function" then
+          code()
+        else
+          error("tempTimer: Code must be a string or a function.")
+        end
+      end
+    end)
+
+    it("should send one command if it is only given one parameter", function()
+      local send = spy.on(_G, "send")
+      sendAll("look")
+      assert.spy(send).was.called(1)
+      assert.spy(send).was.called_with("look", true)
+    end)
+
+    it("should send multiple commands when given multiple string parameters", function()
+      local send = spy.on(_G, "send")
+      local commands = {
+        "get gold from pouch",
+        "buy potion",
+        "put gold in pouch"
+      }
+      sendAll(unpack(commands))
+      assert.spy(send).was.called(#commands)
+      for _,command in ipairs(commands) do
+        assert.spy(send).was.called_with(command, true)
+      end
+    end)
+
+    it("should pass along the final boolean argument to all sends if provided", function()
+      local send = spy.on(_G, "send")
+      sendAll("get gold from pouch", "buy potion", "put gold in pouch", false)
+      assert.spy(send).was.called(3)
+      assert.spy(send).was.called_with("get gold from pouch", false)
+      assert.spy(send).was.called_with("buy potion", false)
+      assert.spy(send).was.called_with("put gold in pouch", false)
+    end)
+
+    it("should pass along the final boolean argument to all sends if provided", function()
+      local send = spy.on(_G, "send")
+      sendAll("get gold from pouch", "buy potion", "put gold in pouch", true)
+      assert.spy(send).was.called(3)
+      assert.spy(send).was.called_with("get gold from pouch", true)
+      assert.spy(send).was.called_with("buy potion", true)
+      assert.spy(send).was.called_with("put gold in pouch", true)
+    end)
+  end)
+
+  describe("permGroup", function()
+
+    describe("success", function()
+      setup(function()
+        _G.oldPermTimer = _G.permTimer
+        _G.oldPermSubstringTrigger = _G.permSubstringTrigger
+        _G.oldPermAlias = _G.permAlias
+        _G.permTimer = function() return 1 end
+        _G.permSubstringTrigger = function() return 1 end
+        _G.permAlias = function() return 1 end
+      end)
+
+      teardown(function()
+        _G.permTimer = _G.oldPermTimer
+        _G.permSubstringTrigger = _G.oldPermSubstringTrigger
+        _G.permAlias = _G.oldPermAlias
+        _G.oldPermTimer = nil
+        _G.oldPermSubstringTrigger = nil
+        _G.oldPermAlias = nil
+      end)
+
+      it("should return true if the timer group was created", function()
+        local permTimer = spy.on(_G, "permTimer")
+        local name = "TestTimer"
+        local parent = "Parent"
+        local successful = permGroup(name, "timer", parent)
+        assert.spy(permTimer).was.called_with(name, parent, 0, "")
+        assert.is_true(successful)
```

Comment:
Should also check that the timer exists with https://wiki.mudlet.org/w/Manual:Mudlet_Object_Functions#exists, right?